src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -413,6 +413,15 @@ protected Task<ImmutableArray<Document>> FindDocumentsWithDeconstructionAsync(Pr
             }, cancellationToken);
         }
 
+        protected Task<ImmutableArray<Document>> FindDocumentsWithAwaitExpressionAsync(Project project, IImmutableSet<Document> documents, CancellationToken cancellationToken)
+        {
+            return FindDocumentsAsync(project, documents, async (d, c) =>
+            {
+                var info = await SyntaxTreeIndex.GetIndexAsync(d, c).ConfigureAwait(false);
+                return info.ContainsAwait;
+            }, cancellationToken);
+        }
```

Comment:
note: we're starting to have a bunch of duplication with the above methods.  consider having a single helper here that takes a ```Func<SyntaxTreeIndex, bool>``` and does all the common work.